### PR TITLE
Fix issues in MetadataAsSource XML documentation comment conversion

### LINKED ISSUE #39877: Decompiled source includes an empty comment above each method
**Version Used**: VS master 29513.77

**Steps to Reproduce**:

1. Open ProjectSystem.sln from https://github.com/dotnet/project-system
2. Navigate ManagedProjectSystemPackage.cs
3. Go To Defition on the `PackageRegistration` attribute (or open any project and Go To Definition on anything from a reference that doesn't have XML comments)

**Expected Behavior**:

Decompilation is shown without comments above members

**Actual Behavior**:

An empty single line comment is displayed, initially collapsed, above members.

This is annoying because it means I have to expand each collapsed section to see if there is any comments for me to read.

### PR DESCRIPTION
**Pull request by siegfriedpammer**
This fixes #39877 

See https://github.com/dotnet/roslyn/issues/39877#issuecomment-575386083 for a detailed description.

TL;DR
1. Only adds a comment above a member, if `DocCommentConverter.ConvertDocCommentToRegularComment` returns anything.
2. Strips leading whitespace before checking whether a line starts with a single-line documentation comment (i.e. `///`).

### CODE REVIEW

**Comment by sharwell** [comment]
Wow, adding a test for this is way more difficult than I expected. I am hoping to get a baseline test completed shortly as part of #24188, and once that is merged we can update this pull request to show how it improves the situation.

**Comment by sharwell** [comment]
📝 I'm doing a manual validation for this so it can be merged sooner.

**Comment by siegfriedpammer** [comment]
If there's anything I can help you with to get this done faster, please tell me.

**Comment by sharwell** [comment]
Yeah this is a big improvement 👍
